File: python/8/11478.py
def main():
    string = input()

    substrings = set()
    for length in range(1, len(string)+1):
        for i in range(len(string) - length + 1):
            substring = string[i:i+length]
            substrings.add(substring)

    print(len(substrings))


if __name__ == '__main__':
    main()

File: rust/src/8/11478.rs
use std::io::stdin;
use std::collections::HashSet;

fn main() {
    let mut input = String::new();
    stdin().read_line(&mut input).unwrap();
    let string = input.trim();

    let mut substrings = HashSet::new();
    for length in 1..=string.len() {
        for i in 0..=string.len() - length {
            let substring = &string[i..i+length];
            substrings.insert(substring);
        }
    }

    println!("{}", substrings.len());
}
